benchmarktests/target_secret_gcp.go

```diff
@@ -0,0 +1,210 @@
+	MaxTTL      string `hcl:"max_ttl,optional"`
+}
+
+type GCPSecretRoleset struct {
+	Name        string   `hcl:"name,optional"`
+	SecretType  string   `hcl:"secret_type,optional"`
+	Project     string   `hcl:"project"`
+	Bindings    string   `hcl:"bindings,optional"`
+	TokenScopes []string `hcl:"token_scopes,optional"`
+}
+
+func (g *GCPTest) ParseConfig(body hcl.Body) error {
+	testConfig := &struct {
+		Config *GCPSecretTestConfig `hcl:"config,block"`
+	}{
+		Config: &GCPSecretTestConfig{
+			GCPConfig:  &GCPSecretConfig{Credentials: os.Getenv(GCPSecretCredentials)},
+			GCPRoleset: &GCPSecretRoleset{Name: "benchmark-roleset", SecretType: GCPAccessTokenType, Bindings: os.Getenv(GCPSecretBindings)},
+		},
+	}
+
+	diags := gohcl.DecodeBody(body, nil, testConfig)
+	if diags.HasErrors() {
+		return fmt.Errorf("error decoding to struct: %v", diags)
+	}
+	g.config = testConfig.Config
+
+	if g.config.GCPRoleset.Project == "" {
+		return fmt.Errorf("GCP project is required")
+	}
+
+	if g.config.GCPRoleset.Bindings == "" {
+		return fmt.Errorf("GCP bindings are required")
+	}
+
+	if g.config.GCPConfig.Credentials == "" {
+		return fmt.Errorf("GCP Credentials are required")
+	}
+
+	return nil
+}
+
+func (g *GCPTest) Target(client *api.Client) vegeta.Target {
+	var url string
+
+	if g.config.GCPRoleset.SecretType == GCPAccessTokenType {
+		url = client.Address() + g.pathPrefix + "/roleset/" + g.roleName + "/token"
+	} else if g.config.GCPRoleset.SecretType == GCPServiceAccountType {
+		url = client.Address() + g.pathPrefix + "/roleset/" + g.roleName + "/key"
+	}
+
+	return vegeta.Target{
+		Method: "GET",
+		URL:    url,
+		Header: g.header,
+	}
+}
+
+func (g *GCPTest) Cleanup(client *api.Client) error {
+	g.logger.Trace(cleanupLogMessage(g.pathPrefix))
+	_, err := client.Logical().Delete(strings.Replace(g.pathPrefix, "/v1/", "/sys/mounts/", 1))
+	if err != nil {
+		return fmt.Errorf("error cleaning up mount: %v", err)
+	}
+	return nil
+}
+
+func (g *GCPTest) GetTargetInfo() TargetInfo {
+	return TargetInfo{
+		method:     GCPSecretTestMethod,
+		pathPrefix: g.pathPrefix,
+	}
+}
+
+func (g *GCPTest) Setup(client *api.Client, mountName string, topLevelConfig *TopLevelTargetConfig) (BenchmarkBuilder, error) {
+	var err error
+	secretPath := mountName
+	g.logger = targetLogger.Named(RedisDynamicSecretTestType)
+
+	if topLevelConfig.RandomMounts {
+		secretPath, err = uuid.GenerateUUID()
+		if err != nil {
+			log.Fatalf("can't create UUID")
+		}
+	}
+
+	config := g.config
+	g.logger.Trace(mountLogMessage("secrets", "gcp", secretPath))
+	setupLogger := g.logger.Named(secretPath)
+
+	err = client.Sys().Mount(secretPath, &api.MountInput{
+		Type: "gcp",
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error mounting gcp: %v", err)
+	}
+
+	// check if the credentials argument should be read from file
+	creds := config.GCPConfig.Credentials
+	if len(creds) > 0 && creds[0] == '@' {
+		contents, err := ioutil.ReadFile(creds[1:])
```

Comment:
```suggestion
		contents, err := os.ReadFile(creds[1:])
```
The ioutil is deprecated and underneath this calls os.ReadFile as of Go 1.16. Since this is new, let's avoid putting in deprecated code where we can.